CONTRIBUTING.md

```diff
@@ -1,29 +1,81 @@
+
+ - [Report a bug](https://github.com/mixxxdj/mixxx/wiki/Reporting%20bugs)
+ - [Test the latest developments](https://github.com/mixxxdj/mixxx/wiki/Testing)
+ - [Translate Mixxx to other languages](https://github.com/mixxxdj/mixxx/wiki/Internationalization)
+ - [Improve the Mixxx manual](https://github.com/mixxxdj/manual#readme)
+ - [Improve the Mixxx website](https://github.com/mixxxdj/website)
+
+If you have some coding skills, or you are willing to learn them you can do even more:
+
+ - [Add a mapping for a new controller](https://github.com/mixxxdj/mixxx/wiki/Contributing%20mappings)
+ - [Fix bugs or program new features](#Your-first-Mixxx-code-contribution)
+
+We welcome programmers of all backgrounds and skill levels, and our community will do their best to help you with any problems you encounter.
+
+## Your first Mixxx code contribution ##
+
+### Compile Mixxx for the first time ###
+Your first goal is to get the Mixxx source code and compile it. The steps to do so depend on your operating system:
+- [Compiling on Linux](https://github.com/mixxxdj/mixxx/wiki/Compiling%20on%20Linux)
+- [Compiling on Windows](https://github.com/mixxxdj/mixxx/wiki/Compiling%20on%20Windows)
+- [Compiling on macOS](https://github.com/mixxxdj/mixxx/wiki/Compiling%20on%20macOS)
+
+### Pick an issue to work on ###
+You might already have in mind that feature you always dreamed of! If that's not the case or you want to start with something easy, you can ask for ideas on [Zulip](https://mixxx.zulipchat.com/#narrow/stream/247620-development-help) or pick an [easy issue](https://bugs.launchpad.net/mixxx/+bugs?field.tag=easy).
+
+### Time to code ###
+Now it's time to write some code!
+We recommend to [set up an IDE](https://github.com/mixxxdj/mixxx/wiki/Developer%20Tools) to assit you in the code writing process.
+
+We use [Git](https://git-scm.com/) for source control. Git is a distributed version control system that allows us to keep track of how our code changes and work together on it. If you are new to Git or source control in general, you can have a look at [Using Git](https://github.com/mixxxdj/mixxx/wiki/Using%20Git) to learn how.
+
+If you have trouble with any of this remember that you can always ask on the [Zulip development help stream](https://mixxx.zulipchat.com/#narrow/stream/247620-development-help).
+
+### Open a Pull Request ###
+Once you finish coding, it's time to show your changes to other Mixxx developers. You do so by [opening a pull request](https://github.com/mixxxdj/mixxx/wiki/Using%20Git#open-a-pull-request).
+
+### Contributor Agreement ###
+Because Mixxx is distributed through the macOS AppStore, in order to distribute your changes with Mixxx, we need a record of your permission to distribute them under Mixxx's license. You'll have to read and sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ).
```

Comment:
```suggestion
In the past Mixxx was distributed in the Mac App Store, which has terms conflicting with Mixxx's license, the GNU GPL (version 2 or later). In case we decide to distribute Mixxx in Apple's App Store again, in order to distribute your changes with Mixxx, we need a record of your permission to distribute them under Mixxx's license. You'll have to read and sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ).
```